code/object/objcollide.cpp

```diff
@@ -1023,7 +1023,7 @@ void obj_find_overlap_colliders(SCP_vector<int> &overlap_list_out, SCP_vector<in
 static SCP_vector<int> sort_list_y;
 static SCP_vector<int> sort_list_z;
 
-void obj_sort_and_collide()
+void obj_sort_and_collide(SCP_vector<int>* Collision_list)
```

Comment:
You might need a `nullptr` check on this. It really shouldn't be needed the way it's used, but you never know. I'll leave it up to you.

But if you do add it I might suggest making the default be `nullptr` rather than pointing to `Collision_sort_list`, then if it's `nullptr` assign `Collision_sort_list` at that time. That makes it a little easier to understand what's going to happen without having to look at another part of the code. That's more of a personal preference/style though.

Reply:
I'm all for more readability, so I'll make that change.